### ISSUE TITLE
Problem loading correct mibs?

### ISSUE THREAD

**Issue by HellStorm666** [librenms/librenms#3562]
I think that librenms isn't using the correct mibs for me.

I get a lot of "No such instance currently exists at this OID" or object.
And
"MIB: registering
snmptranslate -M /opt/librenms/mibs -m all enterprises
This doesnt look like a MIB"

debugs:

```
eltomation@observium:/opt/librenms$ ./discovery.php -h librenms -d -m os
LibreNMS 2015.master Discovery
Version info:
Commit SHA: 39fe65ef94a8d3a9023cd405540c74c7aa92a8ba
DB Schema: 114
PHP: 7.0.4-7ubuntu2.1
MySQL: 10.0.24-MariaDB-7
RRDTool: 1.5.5
SNMP: NET-SNMP version: 5.7.3

DEBUG!

SQL[SELECT version FROM `dbSchema` ORDER BY version DESC LIMIT 1]
SQL[SELECT revision FROM `dbSchema`] DB Schema already up to date.

SQL[SELECT * FROM `devices` WHERE status = 1 AND disabled = 0 AND `hostname` LIKE 'librenms' ORDER BY device_id DESC]
SQL[SELECT * FROM devices_attribs WHERE `device_id` = '4'] librenms 4 linux  (unix)
OS: /usr/bin/snmpget -v2c -c COMMUNITY -Ovq -M /opt/librenms/mibs udp:HOSTNAME:161 SNMPv2-MIB::sysDescr.0Linux observium 4.4.0-22-generic #40-Ubuntu SMP Thu May 12 22:03:46 UTC 2016 x86_64

/usr/bin/snmpget -v2c -c COMMUNITY -Ovqn -M /opt/librenms/mibs udp:HOSTNAME:161 SNMPv2-MIB::sysObjectID.0.*.4.1.8*

| Linux observium 4.4.0-22-generic #40-Ubuntu SMP Thu May 12 22:03:46 UTC 2016 x86_64 | .1.3.6.1.4.1.8072.3.2.10 | /usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -M /opt/librenms/mibs udp:HOSTNAME:161 ENTITY-MIB::entPhysicalMfgName.1No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -m IEEE802dot11-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 dot11manufacturerName.5No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -m GANDI-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 GANDI-MIB::rxCounter.0No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -M /opt/librenms/mibs udp:HOSTNAME:161 HOST-RESOURCES-MIB::hrSystemInitialLoadParameters.0"BOOT_IMAGE=/vmlinuz-4.4.0-22-generic root=/dev/mapper/observium--vg-root ro
"

/usr/bin/snmpget -v2c -c COMMUNITY -OqvU -m CAREL-ug40cdz-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 roomTemp.0No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -M /opt/librenms/mibs udp:HOSTNAME:161 ENTITY-MIB::entPhysicalMfgName.1No Such Object available on this agent at this OID


SQL[SELECT attrib_value FROM devices_attribs WHERE `device_id` = '4' AND `attrib_type` = 'poll_mib' ]
SQL[SELECT attrib_value FROM devices_attribs WHERE `device_id` = '4' AND `attrib_type` = 'poll_mib' ] MIB: registering
snmptranslate -M /opt/librenms/mibs -m all enterprises.8072.3.2.10 2>/dev/null
This doesn't look like a MIB:
MIB: Could not find all::enterprises.8072.3.2.10


SQL[UPDATE `devices` set `last_discovered` =NOW(),`type` ='server',`last_discovered_timetaken` ='2.657' WHERE `device_id` = '4'] Discovered in 2.657 seconds


SQL[INSERT INTO `perf_times` (`type`,`doing`,`start`,`duration`,`devices`)  VALUES ('discover','librenms','1464166077.2986','2.857','1')] ./discovery.php librenms 2016-05-25 10:48:00 - 1 devices discovered in 2.857 secs
MySQL: Cell[1/0s] Row[5/0s] Rows[1/0s] Column[0/0s] Update[1/0.12s] Insert[1/0.12s] Delete[0/0s]
```

```
eltomation@observium:/opt/librenms$ ./poller.php -h librenms -d -r -f -m os
LibreNMS 2015.master Poller
Version info:
Commit SHA: 39fe65ef94a8d3a9023cd405540c74c7aa92a8ba
DB Schema: 114
PHP: 7.0.4-7ubuntu2.1
MySQL: 10.0.24-MariaDB-7
RRDTool: 1.5.5
SNMP: NET-SNMP version: 5.7.3

DEBUG!
Starting polling run:


SQL[SELECT `device_id` FROM `devices` WHERE `disabled` = 0 AND `hostname` LIKE 'librenms' ORDER BY `device_id` ASC]
SQL[SELECT * FROM `devices` WHERE `device_id` = '4']
SQL[SELECT * FROM `vrf_lite_cisco` WHERE `device_id` = '4']
SQL[SELECT * FROM devices_attribs WHERE `device_id` = '4'] librenms 4 linux (unix)

SQL[INSERT INTO `device_perf` (`xmt`,`rcv`,`loss`,`min`,`max`,`avg`,`device_id`,`timestamp`)  VALUES ('3','3','0','0.05','0.06','0.05','4',NOW())] /usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m SNMPv2-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysObjectID.0SNMPv2-SMI::enterprises.8*

/usr/bin/snmpget -v2c -c COMMUNITY -OQnUst -m SNMPv2-MIB:HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysUpTime.0 sysLocation.0 sysContact.0 sysName.0 sysObjectID.0sysUpTime.0 = 33377
sysLocation.0 = ESXi
sysContact.0 = WE <****@eltomation.com>
sysName.0 = observium
sysObjectID.0 = enterprises.8*

/usr/bin/snmpget -v2c -c COMMUNITY -OvQ -m SNMPv2-MIB:HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysDescr.0Linux observium 4.4.0-22-generic #40-Ubuntu SMP Thu May 12 22:03:46 UTC 2016 x86_64

/usr/bin/snmpget -v2c -c COMMUNITY -OQnUst -m HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 snmpEngineTime.0 hrSystemUptime.0snmpEngineTime.0 = 334
hrSystemUptime.0 = 69762016

Using hrSystemUptime (697620s)
[RRD Disabled]RRD[update /opt/librenms/rrd/librenms/uptime.rrd N:697620 --daemon unix:/var/run/rrdcached.sock] Uptime: 8 days, 1h 47m
/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m NET-SNMP-EXTEND-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.8072.1.3.2.3.1.1.6.100.105.115.116.114.111No Such Instance currently exists at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m UCD-SNMP-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.2021.7890.1.3.1.1.6.100.105.115.116.114.111No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m UCD-SNMP-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.2021.7890.1.101.1No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m MIB-Dell-10892 -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.674.10892.1.300.10.1.9.1No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m MIB-Dell-10892 -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.674.10892.1.300.10.1.11.1No Such Object available on this agent at this OID


Hardware: Generic x86 64-bit Version: 4.4.0-22-generic Features:  Serial:
[RRD Disabled]RRD[update /opt/librenms/rrd/librenms/poller-perf.rrd N:11.09 --daemon unix:/var/run/rrdcached.sock] [RRD Disabled]RRD[update /opt/librenms/rrd/librenms/ping-perf.rrd N:0.05 --daemon unix:/var/run/rrdcached.sock] Polled in 11.09 seconds
Updating librenms
Array
(
    [uptime] => 697620
    [last_ping] => Array
        (
            [0] => NOW()
        )

    [last_ping_timetaken] => 0.05
    [last_polled] => Array
        (
            [0] => NOW()
        )

    [last_polled_timetaken] => 11.09
)

SQL[UPDATE `devices` set `uptime` ='697620',`last_ping` =NOW(),`last_ping_timetaken` ='0.05',`last_polled` =NOW(),`last_polled_timetaken` ='11.09' WHERE `device_id` = '4'] UPDATED!

SQL[SELECT * FROM device_groups ORDER BY name]
SQL[SELECT alert_schedule.schedule_id FROM alert_schedule LEFT JOIN alert_schedule_items ON alert_schedule.schedule_id=alert_schedule_items.schedule_id WHERE ( alert_schedule_items.target = '4' ) && NOW() BETWEEN alert_schedule.start AND alert_schedule.end LIMIT 1]
SQL[SELECT * FROM device_groups ORDER BY name]
SQL[SELECT alert_rules.* FROM alert_rules LEFT JOIN alert_map ON alert_rules.id=alert_map.rule WHERE alert_rules.disabled = 0 && ( (alert_rules.device_id = -1 || alert_rules.device_id = '4' ) || alert_map.target = '4'  )]

SQL[INSERT INTO `perf_times` (`type`,`doing`,`start`,`duration`,`devices`,`poller`)  VALUES ('poll','librenms','1464166241.6265','11.24','1','observium\n')] ./poller.php librenms 2016-05-25 10:50:52 - 1 devices polled in 11.24 secs

MySQL: Cell[1/0s] Row[4/0s] Rows[5/0s] Column[0/0s] Update[1/0.07s] Insert[2/0.21s] Delete[0/0s]
```

And if I run these commands to a system instead of the librenms server:

```
eltomation@observium:/opt/librenms$ ./discovery.php -h elto-we -d -m os
LibreNMS 2015.master Discovery
Version info:
Commit SHA: 39fe65ef94a8d3a9023cd405540c74c7aa92a8ba
DB Schema: 114
PHP: 7.0.4-7ubuntu2.1
MySQL: 10.0.24-MariaDB-7
RRDTool: 1.5.5
SNMP: NET-SNMP version: 5.7.3

DEBUG!

SQL[SELECT version FROM `dbSchema` ORDER BY version DESC LIMIT 1]
SQL[SELECT revision FROM `dbSchema`] DB Schema already up to date.

SQL[SELECT * FROM `devices` WHERE status = 1 AND disabled = 0 AND `hostname` LIKE 'elto-we' ORDER BY device_id DESC]
SQL[SELECT * FROM devices_attribs WHERE `device_id` = '1'] elto-we 1 windows
OS: /usr/bin/snmpget -v2c -c COMMUNITY -Ovq -M /opt/librenms/mibs udp:HOSTNAME:161 SNMPv2-MIB::sysDescr.0Hardware: Intel64 Family 6 Model 58 Stepping 9 AT/AT COMPATIBLE - Software: Windows Version 6.1 (Build 7601 Multiprocessor Free)

/usr/bin/snmpget -v2c -c COMMUNITY -Ovqn -M /opt/librenms/mibs udp:HOSTNAME:161 SNMPv2-MIB::sysObjectID.0.*.*.*

| Hardware: Intel64 Family 6 Model 58 Stepping 9 AT/AT COMPATIBLE - Software: Windows Version 6.1 (Build 7601 Multiprocessor Free) | .1.3.6.1.4.1.311.1.1.3.1.1 | /usr/bin/snmpget -v2c -c COMMUNITY -Osqnv -M /opt/librenms/mibs udp:HOSTNAME:161 ENTITY-MIB::entPhysicalMfgName.1No Such Object available on this agent at this OID


SQL[SELECT attrib_value FROM devices_attribs WHERE `device_id` = '1' AND `attrib_type` = 'poll_mib' ]
SQL[SELECT attrib_value FROM devices_attribs WHERE `device_id` = '1' AND `attrib_type` = 'poll_mib' ] MIB: registering
snmptranslate -M /opt/librenms/mibs -m all enterprises.311.1.1.3.1.1 2>/dev/null
This doesn't look like a MIB:
MIB: Could not find all::enterprises.311.1.1.3.1.1


SQL[UPDATE `devices` set `last_discovered` =NOW(),`type` ='laptop',`last_discovered_timetaken` ='2.410' WHERE `device_id` = '1'] Discovered in 2.410 seconds


SQL[INSERT INTO `perf_times` (`type`,`doing`,`start`,`duration`,`devices`)  VALUES ('discover','elto-we','1464166699.6047','2.566','1')] ./discovery.php elto-we 2016-05-25 10:58:22 - 1 devices discovered in 2.566 secs
MySQL: Cell[1/0s] Row[5/0s] Rows[1/0s] Column[0/0s] Update[1/0.08s] Insert[1/0.14s] Delete[0/0s]
eltomation@observium:/opt/librenms$ ./poller.php -h elto-we -d -r -f -m os
LibreNMS 2015.master Poller
Version info:
Commit SHA: 39fe65ef94a8d3a9023cd405540c74c7aa92a8ba
DB Schema: 114
PHP: 7.0.4-7ubuntu2.1
MySQL: 10.0.24-MariaDB-7
RRDTool: 1.5.5
SNMP: NET-SNMP version: 5.7.3

DEBUG!
Starting polling run:


SQL[SELECT `device_id` FROM `devices` WHERE `disabled` = 0 AND `hostname` LIKE 'elto-we' ORDER BY `device_id` ASC]
SQL[SELECT * FROM `devices` WHERE `device_id` = '1']
SQL[SELECT * FROM `vrf_lite_cisco` WHERE `device_id` = '1']
SQL[SELECT * FROM devices_attribs WHERE `device_id` = '1'] elto-we 1 windows

SQL[INSERT INTO `device_perf` (`xmt`,`rcv`,`loss`,`min`,`max`,`avg`,`device_id`,`timestamp`)  VALUES ('3','3','0','0.62','16.4','11.0','1',NOW())] /usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m SNMPv2-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysObjectID.0SNMPv2-SMI::enterprises.*.1.1

/usr/bin/snmpget -v2c -c COMMUNITY -OQnUst -m SNMPv2-MIB:HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysUpTime.0 sysLocation.0 sysContact.0 sysName.0 sysObjectID.0sysUpTime.0 = 1060840
sysLocation.0 = Besturing
sysContact.0 = HE/WE
sysName.0 = Elto-WE.EltomationBV.local
sysObjectID.0 = enterprises.*.1.1

/usr/bin/snmpget -v2c -c COMMUNITY -OvQ -m SNMPv2-MIB:HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 sysDescr.0Hardware: Intel64 Family 6 Model 58 Stepping 9 AT/AT COMPATIBLE - Software: Windows Version 6.1 (Build 7601 Multiprocessor Free)

/usr/bin/snmpget -v2c -c COMMUNITY -OQnUst -m HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB -M /opt/librenms/mibs udp:HOSTNAME:161 snmpEngineTime.0 hrSystemUptime.0snmpEngineTime.0 = No Such Object available on this agent at this OID
hrSystemUptime.0 = 1062731

Using SNMP Agent Uptime (10608s)
[RRD Disabled]RRD[update /opt/librenms/rrd/elto-we/uptime.rrd N:10608 --daemon unix:/var/run/rrdcached.sock] Uptime: 2h 56m 48s
/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m MIB-Dell-10892 -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.674.10892.1.300.10.1.9.1No Such Object available on this agent at this OID

/usr/bin/snmpget -v2c -c COMMUNITY -Oqv -m MIB-Dell-10892 -M /opt/librenms/mibs udp:HOSTNAME:161 .1.3.6.1.4.1.674.10892.1.300.10.1.11.1No Such Object available on this agent at this OID


Hardware: Intel x64 Version: 7 SP1 (NT 6.1) Features: Multiprocessor Serial:
[RRD Disabled]RRD[update /opt/librenms/rrd/elto-we/poller-perf.rrd N:0.961 --daemon unix:/var/run/rrdcached.sock] [RRD Disabled]RRD[update /opt/librenms/rrd/elto-we/ping-perf.rrd N:11.0 --daemon unix:/var/run/rrdcached.sock] Polled in 0.961 seconds
Updating elto-we
Array
(
    [uptime] => 10608
    [last_ping] => Array
        (
            [0] => NOW()
        )

    [last_ping_timetaken] => 11.0
    [last_polled] => Array
        (
            [0] => NOW()
        )

    [last_polled_timetaken] => 0.961
)

SQL[UPDATE `devices` set `uptime` ='10608',`last_ping` =NOW(),`last_ping_timetaken` ='11.0',`last_polled` =NOW(),`last_polled_timetaken` ='0.961' WHERE `device_id` = '1'] UPDATED!

SQL[SELECT * FROM device_groups ORDER BY name]
SQL[SELECT alert_schedule.schedule_id FROM alert_schedule LEFT JOIN alert_schedule_items ON alert_schedule.schedule_id=alert_schedule_items.schedule_id WHERE ( alert_schedule_items.target = '1' ) && NOW() BETWEEN alert_schedule.start AND alert_schedule.end LIMIT 1]
SQL[SELECT * FROM device_groups ORDER BY name]
SQL[SELECT alert_rules.* FROM alert_rules LEFT JOIN alert_map ON alert_rules.id=alert_map.rule WHERE alert_rules.disabled = 0 && ( (alert_rules.device_id = -1 || alert_rules.device_id = '1' ) || alert_map.target = '1'  )]

SQL[INSERT INTO `perf_times` (`type`,`doing`,`start`,`duration`,`devices`,`poller`)  VALUES ('poll','elto-we','1464166716.2156','1.112','1','observium\n')] ./poller.php elto-we 2016-05-25 10:58:37 - 1 devices polled in 1.112 secs

MySQL: Cell[1/0s] Row[4/0s] Rows[5/0s] Column[0/0s] Update[1/0.08s] Insert[2/0.32s] Delete[0/0s]
```

**Comment by exarv** [comment]
I also had some issues enabling the MIB polling and I got the same error:
MIB: Could not find all::enterprises.311.1.1.3.1.1

I've investigated a bit, and it appears that snmptranslate can not find the OID.
After some googling and trial and error, I found that enabling "random access" lookup (-IR) in the snmp-translate() function fixes the "Could not find all::enterprises.xxx" error and MIB associations pop up on the librenms web interface.

I've created a pull request for the fix. #3576
